Sources/SwiftGenKit/Parsers/JSON/JSONParser.swift

```diff
@@ -0,0 +1,42 @@
+//
+// SwiftGenKit
+// Copyright © 2019 SwiftGen
+// MIT Licence
+//
+
+import Foundation
+import PathKit
+
+public enum JSON {
+  public enum ParserError: Error, CustomStringConvertible {
+    case invalidFile(path: Path, reason: String)
+
+    public var description: String {
+      switch self {
+      case .invalidFile(let path, let reason):
+        return "Unable to parse file at \(path). \(reason)"
+      }
+    }
+  }
+
+  // MARK: JSON File Parser
+
+  public class Parser: SwiftGenKit.Parser {
+    private let options: ParserOptionValues
+    var files: [File] = []
+    public var warningHandler: Parser.MessageHandler?
+
+    public required init(options: [String: Any] = [:], warningHandler: Parser.MessageHandler? = nil) throws {
+      self.options = try ParserOptionValues(options: options, available: Parser.allOptions)
+      self.warningHandler = warningHandler
+    }
+
+    public class var defaultFilter: String {
+      "[^/]\\.(?i:json)$"
+    }
```

Comment:
As mentioned below, this can become a `let` now that we don't need to override:
```suggestion
    public static let defaultFilter = "[^/]\\.(?i:json)$"
```